packages/core/src/client.ts

```diff
@@ -0,0 +1,179 @@
+	GatewayMessageDeleteBulkDispatchData,
+	GatewayMessageDeleteDispatchData,
+	GatewayMessageReactionAddDispatchData,
+	GatewayMessageReactionRemoveAllDispatchData,
+	GatewayMessageReactionRemoveDispatchData,
+	GatewayMessageReactionRemoveEmojiDispatchData,
+	GatewayMessageUpdateDispatchData,
+	GatewayPresenceUpdateDispatchData,
+	GatewayReadyDispatchData,
+	GatewayResumedDispatch,
+	GatewayStageInstanceCreateDispatchData,
+	GatewayStageInstanceDeleteDispatchData,
+	GatewayStageInstanceUpdateDispatchData,
+	GatewayThreadCreateDispatchData,
+	GatewayThreadDeleteDispatchData,
+	GatewayThreadListSync,
+	GatewayThreadMembersUpdateDispatchData,
+	GatewayThreadMemberUpdateDispatchData,
+	GatewayThreadUpdateDispatchData,
+	GatewayTypingStartDispatchData,
+	GatewayUserUpdateDispatchData,
+	GatewayVoiceServerUpdateDispatchData,
+	GatewayVoiceStateUpdateDispatchData,
+	GatewayWebhooksUpdateDispatchData,
+} from 'discord-api-types/v10';
+import { API } from './api/index.js';
+
+export interface IntrinsicProps {
+	/**
+	 * The rest API
+	 */
+	api: API;
+	/**
+	 * The id of the shard that emitted the event
+	 */
+	shardId: number;
+}
+
+export interface WithIntrinsicProps<T> extends IntrinsicProps {
+	data: T;
+}
+
+export interface MappedEvents {
+	[GatewayDispatchEvents.ChannelCreate]: [WithIntrinsicProps<GatewayChannelCreateDispatchData>];
+	[GatewayDispatchEvents.ChannelDelete]: [WithIntrinsicProps<GatewayChannelDeleteDispatchData>];
+	[GatewayDispatchEvents.ChannelPinsUpdate]: [WithIntrinsicProps<GatewayChannelPinsUpdateDispatchData>];
+	[GatewayDispatchEvents.ChannelUpdate]: [WithIntrinsicProps<GatewayChannelUpdateDispatchData>];
+	[GatewayDispatchEvents.GuildBanAdd]: [WithIntrinsicProps<GatewayGuildBanAddDispatchData>];
+	[GatewayDispatchEvents.GuildBanRemove]: [WithIntrinsicProps<GatewayGuildBanRemoveDispatchData>];
+	[GatewayDispatchEvents.GuildCreate]: [WithIntrinsicProps<GatewayGuildCreateDispatchData>];
+	[GatewayDispatchEvents.GuildDelete]: [WithIntrinsicProps<GatewayGuildDeleteDispatchData>];
+	[GatewayDispatchEvents.GuildEmojisUpdate]: [WithIntrinsicProps<GatewayGuildEmojisUpdateDispatchData>];
+	[GatewayDispatchEvents.GuildIntegrationsUpdate]: [WithIntrinsicProps<GatewayGuildIntegrationsUpdateDispatchData>];
+	[GatewayDispatchEvents.GuildMemberAdd]: [WithIntrinsicProps<GatewayGuildMemberAddDispatchData>];
+	[GatewayDispatchEvents.GuildMemberRemove]: [WithIntrinsicProps<GatewayGuildMemberRemoveDispatchData>];
+	[GatewayDispatchEvents.GuildMemberUpdate]: [WithIntrinsicProps<GatewayGuildMemberUpdateDispatchData>];
+	[GatewayDispatchEvents.GuildMembersChunk]: [WithIntrinsicProps<GatewayGuildMembersChunkDispatchData>];
+	[GatewayDispatchEvents.GuildRoleCreate]: [WithIntrinsicProps<GatewayGuildRoleModifyDispatchData>];
+	[GatewayDispatchEvents.GuildRoleDelete]: [WithIntrinsicProps<GatewayGuildRoleDeleteDispatchData>];
+	[GatewayDispatchEvents.GuildRoleUpdate]: [WithIntrinsicProps<GatewayGuildRoleModifyDispatchData>];
```

Comment:
Does -types not define `GatewayGuildRoleUpdateDispatchData` even if it would be an alias of `GatewayGuildRoleModifyDispatchData`?